plugins/deep-link/src/lib.rs

```diff
@@ -304,12 +305,28 @@ mod imp {
 
                 if let Ok(mut desktop_file) = ini::Ini::load_from_file(&target_file) {
                     if let Some(section) = desktop_file.section_mut(Some("Desktop Entry")) {
-                        // it's ok to remove it - we only write to the file if it's missing
-                        // and in that case we include old_mimes
                         let old_mimes = section.remove("MimeType").unwrap_or_default();
+                        let mut change = false;
 
+                        // if the mime type is not present, append it to the list
                         if !old_mimes.split(';').any(|mime| mime == mime_type) {
                             section.append("MimeType", format!("{mime_type};{old_mimes}"));
+                            change = true;
+                        } else {
+                            section.insert("MimeType".to_string(), old_mimes);
+                        }
+
+                        // if the exec command doesnt match, update to the new one
+                        let old_exec = section.remove("Exec").unwrap_or(qualified_exec.clone());
```

Comment:
```suggestion
                        let old_exec = section.remove("Exec").unwrap_or_default();
```
I think something like this makes more sense here just in case that Exec is empty or doesn't exist (pretty unrealistic)